File: 010.funcao/java_src/funcao1.5.java
package funcao15;

public class Funcao15 {

		public static void imprimeVariavel () {
			int a;
			a = 5;
			
			for( int i = 0; i<3; i++ ){
				System.out.print("Valor de a: ");
				System.out.println(a);
				a = a + 1;
				System.out.print("Iteracao: ");
				System.out.println(i);
			}
			
			System.out.print("Valor de a: ");
			System.out.println(a);
			
		}

		public static void main(String[] Args) {
			imprimeVariavel();
		}
}

File: 012.jarakirefactor/test/ebin/funcao15.erl
-module(funcao15).

-compile(export_all).

-import(loop, [for/3, while/2]).

imprimeVariavel() ->
    st:get_new_stack(imprimeVariavel),
    st:put_value({imprimeVariavel, "a"}, {int, 5}),
    begin
      st:put_value({imprimeVariavel, "i"}, {int, 0}),
      for(fun () -> st:get_value(imprimeVariavel, "i") < 3
	  end,
	  fun () ->
		  st:put_value({imprimeVariavel, "i"},
			       {int, st:get_value(imprimeVariavel, "i") + 1})
	  end,
	  fun () ->
		  io:format("~s", ["Valor de a: "]),
		  io:format("~p~n", [st:get_value(imprimeVariavel, "a")]),
		  st:put_value({imprimeVariavel, "a"},
			       {int, st:get_value(imprimeVariavel, "a") + 1}),
		  io:format("~s", ["Iteracao: "]),
		  io:format("~p~n", [st:get_value(imprimeVariavel, "i")])
	  end),
      st:delete(imprimeVariavel, "i")
    end,
    io:format("~s", ["Valor de a: "]),
    io:format("~p~n", [st:get_value(imprimeVariavel, "a")]).

main(V_Args) ->
    st:new(),
    st:get_new_stack(main),
    st:put_value({main, "Args"}, {'String', V_Args}),
    st:return_function(fun () -> imprimeVariavel() end,
		       imprimeVariavel, []),
    st:get_old_stack(main),
    st:destroy().

